scipy/stats/tests/test_qmc.py

```diff
@@ -570,6 +591,20 @@ def test_warning(self):
             seed = np.random.RandomState(12345)
             qmc.LatinHypercube(1, seed=seed, optimization="toto")
 
+        with pytest.raises(ValueError, match=r"n is not the square of a"
+                                             r" prime number"):
+            engine = qmc.LatinHypercube(d=2, orthogonal=True, seed=seed)
+            engine.random(16)
+
+        with pytest.raises(ValueError, match=r"n is not the square of a"
+                                             r" prime number"):
+            engine = qmc.LatinHypercube(d=2, orthogonal=True, seed=seed)
+            engine.random(5)  # because int(sqrt(5)) would result in 2
+
+        with pytest.raises(ValueError, match=r"d is too large"):
```

Comment:
Perhaps it is more clear to warn that `n` is too small. There was no problem creating `engine = qmc.LatinHypercube(d=5, orthogonal=True, seed=seed)`.

Reply:
Good point, I will change this. In the end it's otherwise confusing as the user wants a specific dim and can most of the time accommodate for deviations on the number of samples.